File: app/src/main/java/com/selfmate/mes/selfmate/Models/TypeConverters/DateTypeConverter.java
package com.selfmate.mes.selfmate.Models.TypeConverters;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 * Created by vicki_mes on 11/16/2017.
 */


public class DateTypeConverter {

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/selfmate/mes/selfmate/models/tc/DateTypeConverter.kt
package com.selfmate.mes.selfmate.models.tc

import android.arch.persistence.room.TypeConverter
import java.util.*

/**
 * Created by vicki_mes on 11/16/2017.
 */


class DateTypeConverter {

    @TypeConverter
    fun toDate(timestamp: Long?): Date? = if (timestamp == null) null else Date(timestamp)

    @TypeConverter
    fun toTimestamp(date: Date?): Long? = date?.time
}

